client/src/components/Agents.js

```diff
@@ -0,0 +1,85 @@
+import { useContext, useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import AuthContext from '../AuthContext';
+
+function Agents() {
+    const [agents, setAgents] = useState([]);
+
+    const auth = useContext(AuthContext);
+
+    const getAgents = (token) => {
+
+        const init = {
+            headers: {
+                'Authorization': `Bearer ${token}`
+            }
+        }
+
+        fetch('http://localhost:8080/api/agent', init)
+            .then(response => response.json())
+            .then(data => setAgents(data))
+            .catch(error => console.log(error));
+    };
+
+    useEffect(() => {
+        getAgents(auth.user.token);
+    }, [auth.user.token]);
+
+
+    const agentDeleteClickHandler = (agentId) => {
+
+        const init = {
+          method: 'DELETE', // GET by default
+          headers: {
+            'Authorization': `Bearer ${auth.user.token}`
+          }
+        };
+    
+        fetch(`http://localhost:8080/api/agent/${agentId}`, init)
+          .then(response => {
+            if (response.status === 204) {
+              getAgents(auth.user.token);
+            } else if (response.status === 404) {
+              Promise.reject(`ToDo ID ${agentId} not found`);
+            } else {
+              Promise.reject('Something unexpected went wrong :)');
+            }
+          })
+          .catch(error => console.log(error));
+      };
```

Comment:
Separate page needed for the "/agents/delete/:id" path, I used a Link to the separate component with a deletion confirmation on it.